dynamic_instrumentation/src/instrument.rs

```diff
@@ -203,14 +205,19 @@ impl<'tcx> Visitor<'tcx> for InstrumentationAdder<'_, 'tcx> {
                 self.loc(location, store_fn)
                     .arg_var(base_dest)
                     .source(&remove_outer_deref(dest, self.tcx()))
+                    .debug_mir(location)
                     .add_to(self);
 
                 if is_region_or_unsafe_ptr(value_ty) {
-                    location.statement_index += 1;
-                    self.loc(location, store_value_fn)
+                    let instrumentation_location = Location {
+                        statement_index: location.statement_index + 1,
+                        ..location
+                    };
```

Comment:
Can we add a `Location::inc` method?

Reply:
Fixed in https://github.com/immunant/c2rust/pull/566/commits/d1429553d6eff8d2b89968c06666e06577344cd3 (used the existing `Location::successor_within_block`, which does exactly the same thing).